src/prime_rl/trainer/rl/loss.py

```diff
@@ -134,3 +132,62 @@ def shift_logits(logits: Float[Tensor, "batch seq vocab"]) -> Float[Tensor, "bat
     zeros = torch.zeros(batch, 1, vocab, device=logits.device, dtype=logits.dtype)  # (batch, 1, vocab)
     logits = torch.cat([zeros, logits], dim=1)  # (batch, seq, vocab)
     return logits
+
+
+def compute_packed_sequence_loss(
+    logprobs, old_logprobs, advantages, loss_mask, response_lengths, loss_config, loss_scale
+):
```

Comment:
also lets make this function take positions_ids and compute the resp lenght inside it